README.md

```diff
@@ -2,50 +2,89 @@
 associated crates.
-Note that this is just basic introduction, not a comprehensive developer guide.
-It doesn't teach about security or other important practices - it'd require whole book, maybe even multiple books!
+Note that this is just basic a introduction, not a comprehensive developer guide.
+It doesn't teach about security or other important practices - which would require a whole book, maybe even multiple books!
 
-The code has regtest hard-coded. This would be normally a bad practice if this wasn't a demo.
+The code has regtest hard-coded, hard-coding the network should not be done in production code.
 
 ## Tutorial
 
 The `master` branch contains only the skeleton which you can use to implement the wallet yourself.
 Implementing your own wallet will give you much better insight into Bitcoin and its Rust library.
-The skeleton provides configuration and database support.
+The skeleton provides configuration and database support plus some minimal helper functions.
+
+
+### Overview
+
+0. Install/configure bitcoind regtest node (see below).
+1. Configure `pico-bitcoin-wallet` (see below).
+2. Read the code in `lib.rs`
+3. Implement the `address` command.
+4. Implement the `scan` command.
+5. Implement the `send` command.
+
+
+Solutions are provide per-step as git branches.
 
-### Usage
+Happy hacking!
+
+### Configure bitcoind
 
-Install and run bitcoind in regtest mode.
 [Cryptoanarchy Debian Repository](https://github.com/debian-cryptoanarchy/cryptoanarchy-deb-repo-builder) may help.
-If you install `bitcoin-timechain-regtest` you don't need any further configuration.
+
 Otherwise configure the wallet in `~/.config/pico-bitcoin-wallet/config.toml`.
 
-Create a wallet in bitcoind and mine some coins (mine at least 101 blocks so that coins are mature).
+Example minimal `bitcoin.conf`
+```
+chain=regtest
+# Password for JSON-RPC connections
+rpcpassword=pass
+# Username for JSON-RPC connections
+rpcuser=admin
+```
+
+### Configure pico-bitcoin-wallet
 
-You should implement these commands:
+If you used `bitcoin-timechain-regtest` from `cryptoanarchy` you don't need any further
+configuration.
 
-* `address` - returns the address of the wallet - create it from private key.
-* `scan` - scans the blocks to find matching transactions, should also tell the user how many coins were found.
-* `send` - creates a transaction that sends a given amount to the address provided in the input, bonus: implement BIP21
+Otherwise create `~/.config/pico-bitcoin-wallet/config.toml` (exact location will vary depending on your OS).
 
-First version can be really very lazy if you like. E.g. require scan after sending transaction.
+```toml
+bitcoind_uri = "http://localhost:18443"
+bitcoind_username = "admin"
+bitcoind_password = "pass"
+```
+
+### Mine coins
+
+After you have completed step 3 (implementing the `address` command) you can use `bitcoin-cli
+generatetoaddress 101 <address>` to mine some coins. This is the only `bitcoin-cli` command you need
```

Comment:
To do this correctly the wallet would have to handle coinbase transactions correctly. Which seems too much for a simple wallet, so I prefer mine to a Core wallet and then send.

Reply:
Oh good point, I ran into coinbase tx errors while doing this, had not found a minimal way to document that. I like your idea, will update.